src/libs/actions/PolicyConnections.ts

```diff
@@ -0,0 +1,83 @@
+import Onyx from 'react-native-onyx';
+import type {OnyxUpdate} from 'react-native-onyx';
+import * as API from '@libs/API';
+import type UpdateConnectionConfigParams from '@libs/API/parameters/UpdateConnectionConfigParams';
+import {WRITE_COMMANDS} from '@libs/API/types';
+import * as ErrorUtils from '@libs/ErrorUtils';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {ConnectionConfig, ConnectionName} from '@src/types/onyx/Policy';
+
+function updatePolicyConnectionConfig({
+    policyID,
+    connectionName,
+    updatedField,
+    updatedPartialConfig,
+    originalPartialConfig,
+}: {
+    policyID: string;
+    connectionName: ConnectionName;
+    updatedField: string;
+    updatedPartialConfig: Partial<ConnectionConfig>;
+    originalPartialConfig: Partial<ConnectionConfig>;
+}) {
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                pendingFields: {
+                    [updatedField]: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+                },
+                connections: {
+                    [connectionName]: {
+                        config: updatedPartialConfig,
+                    },
+                },
+            },
+        },
+    ];
+
+    const successData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                pendingFields: {
+                    [updatedField]: null,
+                },
+            },
+        },
+    ];
+
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                pendingFields: {
+                    [updatedField]: null,
+                },
+                errorFields: {
+                    [updatedField]: ErrorUtils.getMicroSecondOnyxError('workspace.connection.error.genericUpdate'),
+                },
+                connections: {
+                    [connectionName]: {
+                        config: originalPartialConfig,
+                    },
+                },
+            },
+        },
+    ];
+
+    const parameters: UpdateConnectionConfigParams = {
+        policyId: policyID,
+        connectionName,
+        config: updatedPartialConfig,
```

Comment:
@hayata-suenaga just to clarify - based on the high level doc - why do we send config instead of exact field which we currently switch off/on? Is it some new requirement from API>
Not sure if also pendingFields are defined correctly - because Policy itself is not having fields like syncClasses, syncLocations and etc - TS will fire errors during usage inside Import pages
Maybe i did it wrong - but can you check my implementation of this one
https://github.com/Expensify/App/pull/38400/files#diff-cd0dfd9f8af5bc7cd33ff4f0337ef70574d42424ceecd7692b193348d3d8ecd4
Just to be sure that we following correct way to build this flow

Reply:
The backend command `UpdatePolicyConnectionConfiguration` ended up being a bit different from what we laid out in the design doc , mainly because we thought it was not worth it the complexity of editing the exact field. Currently it needs the following parameters (so the implementation here is wrong):

- `settingName`: the key inside `config` that you are trying to change. This can't be a nested key.
- `settingValue`: the value for the key `settingName`
- `connectionName`
- `policyID`

So, if the `policy.connections` look like:


```json
{
    "quickbooksOnline": {
        "lastSync": {
            "successfulDate": "2024-02-13T02:52:24+0000",
            "isSuccessful": true,
            "errorDate": "",
            "source": "EXPENSIFYWEB"
        },
        "data": { ... },
        "config": {
            "receivableAccount": {
                "glCode": "",
                "name": "Accounts Receivable",
                "currency": "USD",
                "id": "53"
            },
            "syncClasses": "TAG",
            "credentials": { ... },
            "syncCustomers": "TAG",
            "companyName": "Expensify",
            "autoCreateVendor": true,
            "nonReimbursableBillDefaultVendor": "138",
            "enableNewCategories": true,
            "syncLocations": "NONE",
            "nonReimbursableExpensesAccount": {
                "glCode": "",
                "name": "Accounts Payable",
                "currency": "USD",
                "id": "54"
            },
            "export": {
                "exporter": "admin@qbo.com"
            },
            "reimbursableExpensesAccount": {
                "glCode": "",
                "name": "Accounts Payable",
                "currency": "USD",
                "id": "54"
            },
            "syncItems": false,
            "lastConfigurationTime": 1708460337446,
            "realmId": "1423559500",
            "collectionAccountID": "",
            "markChecksToBePrinted": false,
            "autoSync": {
                "jobID": "5736989436320711907",
                "enabled": true
            },
            "reimbursableExpensesExportDestination": "bill",
            "syncPeople": false,
            "hasChosenAutoSyncOption": true,
            "nonReimbursableExpensesExportDestination": "bill",
            "exportDate": "LAST_EXPENSE",
            "syncTax": false,
            "reimbursementAccountID": ""
        }
    }
}
```

then possible pairs of `settingName`, `settingValue` could be:

```
settingName: "syncPeople"
settingValue: false
```

```
settingName: "reimbursableExpensesAccount"
settingValue: {"glCode": "", "name": "Accounts Payable", "currency": "USD", "id": "54"}
```

```
settingName: "export"
settingValue: {"exporter": "admin@qbo.com"}
```

`settingName` can't be a chain like `export.exporter`

Reply:
fixed https://github.com/Expensify/App/pull/38006/commits/14ae20a2d427e1f360d7638ef804685aeb797624 🙇